client/src/views/Dashboard.vue

```diff
@@ -150,26 +157,33 @@ function setSearchQuery(searchSentence: string) {
   searchQuery.value = searchSentence
 }
 
-function addTodo(todo: Todo) {
-  const todoCopy = reactive<Todo>({
-    title: todo.title,
-    priority: todo.priority,
-    description: todo.description,
-    isChecked: todo.isChecked,
-    date: todo.date
-  })
-  todos.unshift(todoCopy)
+async function addTodo(todo: Todo) {
+  try {
+    await addNewTodo(
+      todo.title,
+      todo.priority,
+      todo.description,
+      todo.isChecked,
+      todo.date
+    )
+    fetchAndAddTodos()
```

Comment:
you don't need to fetch each todo after the add, you just need to append to the list